src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/ciagmigration/mapper/FutureWorkInterestsMigrationMapper.kt

```diff
@@ -0,0 +1,78 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.mapper
+
+import org.mapstruct.AfterMapping
+import org.mapstruct.Mapper
+import org.mapstruct.Mapping
+import org.mapstruct.MappingTarget
+import org.mapstruct.NullValueMappingStrategy
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.jpa.entity.FutureWorkInterestsMigrationEntity
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.jpa.entity.WorkInterestMigrationEntity
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.resource.model.WorkInterestDetail
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.ciagmigration.resource.model.WorkInterestsResponse
+import java.time.Instant
+
+@Mapper(
+  uses = [
+    LocalDateTimeMapper::class,
+  ],
+  nullValueIterableMappingStrategy = NullValueMappingStrategy.RETURN_DEFAULT,
+)
+abstract class FutureWorkInterestsMigrationMapper {
+
+  fun toFutureWorkInterestsMigrationEntity(
```

Comment:
I've added this type of method to each mapper class because the generated one (which is now based on `createFutureWorkInterestsMigrationEntity()` below) had this null check at the beginning of each one:

```
if ( prisonId == null && workInterests == null ) {
   return null;
}
```
An or (`||`) would have been better, but on the basis `prisonId` is mandatory in the API, but workInterests isn't, I've added an if statement here to check only the latter.